controllers/articles-controller.js

```diff
@@ -13,7 +15,39 @@ function getArticleById(req, res, next) {
 }
 
 function getArticles(req, res, next) {
-  selectArticles()
+  selectTopics()
+    .then((topics) => {
+      return Promise.resolve(
+        topics.map((topic) => {
+          return topic.slug;
+        })
+      );
+    })
+    .then((topicList) => {
+      const topic = req.query.topic;
+      const sort_by = req.query.sort_by ? req.query.sort_by : "created_at";
+      const order = req.query.order ? req.query.order.toUpperCase() : "DESC";
+      const validSortBy = [
+        "article_id",
+        "title",
+        "topic",
+        "author",
+        "body",
+        "created_at",
+        "votes",
+        "comment_count",
+        "article_img_url",
+      ];
+      if (
+        (topicList.includes(topic) || !topic) &&
+        validSortBy.includes(sort_by) &&
+        ["ASC", "DESC"].includes(order)
+      ) {
+        return selectArticles(topic, sort_by, order);
+      } else {
+        return Promise.reject({ status: 404, msg: "Query not available" });
+      }
```

Comment:
I feel inclined to suggest moving this logic to check for valid sortBy and orderBy to the model function, in terms of the division of labour between controllers / models it feels to me the more natural place for it to be 

Models handle everything to do with the database and the queries, the controller doesn't care if it's valid it just handles the request and the response

Reply:
Nice suggestion - thanks! Created a util function in models for the time being, although long-term this might belong in a utils folder? Will think about this!